### ISSUE TITLE
RSJob State does not reflect actual state of job

### ISSUE THREAD

**Issue by oldlost** [proxb/PoshRSJob#95]
When creating RSJobs, all objects begin with State of Running even though the actual state should be NotStarted. Only when a job actually begins execution should its state be set to Running.

The other states appear to be correct.

**Comment by proxb** [comment]
Thanks!

**Comment by proxb** [comment]
I've solved this issue and just need to implement it within the module. https://learn-powershell.net/2016/12/04/tracking-the-state-of-powershell-runspaces-in-a-runspacepool/

**Comment by proxb** [comment]
@oldlost This should be fixed now, please test and let me know the results.

**Comment by platta** [comment]
@proxb I just grabbed the latest version today. When I call `Start-RSJob` all the jobs are in a state of `NotStarted` and they remain in that state indefinitely, even though the job itself executes (I can see the downstream actions the jobs are taking). Wondering if that could be an issue introduced by the change made for this issue.

I'm going to do some digging tomorrow to see if I can provide more information, but I wanted to drop a note in case you have an ah-ha moment and already know what is happening.

**Comment by proxb** [comment]
@platta Thanks for the heads up on this. I'm going to also take a look and see if I can duplicate the issue. Just out of curiosity, what version of PowerShell are you running?

**Comment by proxb** [comment]
i did a very simple demo to see what would happen and it appears to be working as expected on PowerShell V5 on Windows 10.

```powershell
PS C:\WINDOWS\system32> 1..10|ssj {start-sleep 10}

Id       Name                 State           HasMoreData  HasErrors    Command
--       ----                 -----           -----------  ---------    -------
1        Job1                 Running         False        False        start-sleep 10
2        Job2                 Running         False        False        start-sleep 10
3        Job3                 Running         False        False        start-sleep 10
4        Job4                 Running         False        False        start-sleep 10
5        Job5                 Running         False        False        start-sleep 10
6        Job6                 NotStarted      False        False        start-sleep 10
7        Job7                 NotStarted      False        False        start-sleep 10
8        Job8                 NotStarted      False        False        start-sleep 10
9        Job9                 NotStarted      False        False        start-sleep 10
10       Job10                NotStarted      False        False        start-sleep 10
```

I'll try some other tests to see what happens.

**Comment by platta** [comment]
The first time it happened I was on 2008 R2 using V4. This morning I was able to reproduce it on Windows 10 using V5. Both those times I had downloaded the source through a browser from the GitHub page. Using Install-Module or the download link in the project's readme both gave me a working example.

Here's the fun part - Git diff shows that all versions of the module are identical, so after all that testing I re-extracted the original zip that gave me "broken" behavior. Now it works. I can no longer reproduce the issue. I replaced the module with a copy downloaded using the link in the readme on the 2008 R2 machine and now that one works too.

I give up!
